### PR TITLE
Add Makefile, Changelog, AGPL, update gitignore

### PR DESCRIPTION
**Pull request by VicDeo**
- Add makefile
- add changelog 
- add COPYING (the source is https://github.com/owncloud/core/blob/master/COPYING )
- update gitignore

### CODE REVIEW

**Comment by VicDeo** [comment]
@PVince81 I've noticed that our default  `make` action for a number of apps is `make clean` 
Is it intentional?

**Comment by PVince81** [comment]
@VicDeo not intentional... need to have "all" as the first rule in the Makefile to make that the default

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+sign_skip_msg="Skipping signing, either no key and certificate found in $(private_key) and $(certificate) or occ can not be found at $(occ)"
+
+ifneq (,$(wildcard $(private_key)))
+ifneq (,$(wildcard $(certificate)))
+ifneq (,$(wildcard $(occ)))
+	CAN_SIGN=true
+endif
+endif
+endif
+
```
Comment (PVince81) [inline Makefile]: let's add `all: appstore` here to make it a default rule

Comment (VicDeo) [reply Makefile]: @PVince81 added

**Pull request merged**